flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTriggeringTest.java

```diff
@@ -289,29 +292,29 @@ public void testStopPeriodicScheduler() throws Exception {
 			failureManager);
 
 		// Periodic
+		CompletableFuture<CompletedCheckpoint> periodicTriggerResult = coord.triggerCheckpoint(
+			System.currentTimeMillis(),
+			CheckpointProperties.forCheckpoint(CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION),
+			null,
+			true,
+			false);
 		try {
-			coord.triggerCheckpoint(
-				System.currentTimeMillis(),
-				CheckpointProperties.forCheckpoint(CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION),
-				null,
-				true,
-				false);
+			periodicTriggerResult.get();
 			fail("The triggerCheckpoint call expected an exception");
-		} catch (CheckpointException e) {
-			assertEquals(CheckpointFailureReason.PERIODIC_SCHEDULER_SHUTDOWN, e.getCheckpointFailureReason());
+		} catch (ExecutionException e) {
+			assertTrue(e.getCause() instanceof CheckpointException);
+			assertEquals(CheckpointFailureReason.PERIODIC_SCHEDULER_SHUTDOWN,
+				((CheckpointException) e.getCause()).getCheckpointFailureReason());
 		}
 
 		// Not periodic
-		try {
-			coord.triggerCheckpoint(
-				System.currentTimeMillis(),
-				CheckpointProperties.forCheckpoint(CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION),
-				null,
-				false,
-				false);
-		} catch (CheckpointException e) {
-			fail("Unexpected exception : " + e.getCheckpointFailureReason().message());
-		}
+		CompletableFuture<CompletedCheckpoint> nonPeriodicTriggerResult = coord.triggerCheckpoint(
+			System.currentTimeMillis(),
+			CheckpointProperties.forCheckpoint(CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION),
+			null,
+			false,
+			false);
+		assertFalse(nonPeriodicTriggerResult.isCompletedExceptionally());
```

Comment:
ditto about waiting for the future to complete? 
```
assertFalse(nonPeriodicTriggerResult.isCompletedExceptionally());
```
 ->
```
nonPeriodicTriggerResult.get();
```
?

Reply:
Ditto.